pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -29,26 +29,27 @@ import (
 
 // PerconaXtraDBClusterSpec defines the desired state of PerconaXtraDBCluster
 type PerconaXtraDBClusterSpec struct {
-	Platform               version.Platform                     `json:"platform,omitempty"`
-	CRVersion              string                               `json:"crVersion,omitempty"`
-	Pause                  bool                                 `json:"pause,omitempty"`
-	SecretsName            string                               `json:"secretsName,omitempty"`
-	VaultSecretName        string                               `json:"vaultSecretName,omitempty"`
-	SSLSecretName          string                               `json:"sslSecretName,omitempty"`
-	SSLInternalSecretName  string                               `json:"sslInternalSecretName,omitempty"`
-	LogCollectorSecretName string                               `json:"logCollectorSecretName,omitempty"`
-	TLS                    *TLSSpec                             `json:"tls,omitempty"`
-	PXC                    *PXCSpec                             `json:"pxc,omitempty"`
-	ProxySQL               *ProxySQLSpec                        `json:"proxysql,omitempty"`
-	HAProxy                *HAProxySpec                         `json:"haproxy,omitempty"`
-	PMM                    *PMMSpec                             `json:"pmm,omitempty"`
-	LogCollector           *LogCollectorSpec                    `json:"logcollector,omitempty"`
-	Backup                 *PXCScheduledBackup                  `json:"backup,omitempty"`
-	UpdateStrategy         appsv1.StatefulSetUpdateStrategyType `json:"updateStrategy,omitempty"`
-	UpgradeOptions         UpgradeOptions                       `json:"upgradeOptions,omitempty"`
-	AllowUnsafeConfig      bool                                 `json:"allowUnsafeConfigurations,omitempty"`
-	Unsafe                 UnsafeFlags                          `json:"unsafeFlags,omitempty"`
-	VolumeExpansionEnabled bool                                 `json:"enableVolumeExpansion,omitempty"`
+	Platform                version.Platform                     `json:"platform,omitempty"`
+	CRVersion               string                               `json:"crVersion,omitempty"`
+	Pause                   bool                                 `json:"pause,omitempty"`
+	SecretsName             string                               `json:"secretsName,omitempty"`
+	GeneratedSecretsOptions *GeneratedSecretsOptions             `json:"generatedSecretsOptions,omitempty"`
```

Comment:
Given that the configuration we introduce is related to the passwords that are provisioned through secrets and not for the secrets themselves, maybe we could adjust the naming of the new options accordingly. Some options:

  - PasswordGenerationOptions - Most accurate and clear
  - GeneratedPasswordOptions - Also clear and descriptive
  - PasswordOptions - Simpler but still clear

 `PasswordGenerationOptions` is the most suitable one IMO.